src/timeout.ts

```diff
@@ -0,0 +1,100 @@
+import { clearTimeout, setTimeout } from 'timers';
+
+import { MongoInvalidArgumentError } from './error';
+import { noop } from './utils';
+
+/** @internal */
+export class TimeoutError extends Error {
```

Comment:
The only place this is thrown is in the `Timeout` class, and we immediately catch it and throw a different error ( this error is internal ).  

Why use an error at all?  Could we just throw a string or something else?

Reply:
Good point. I possibly misunderstood the [spec](https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.md#errors) here. I confused the error we throw to signify a timeout has occurred in this class with the error that we'll have to wrap the server selection timeout/waitqueue timeout error with.

I'm open to just throwing a string here. @nbbeeken thoughts?

Reply:
We can just throw anything since we should/need to always convert the error. However, I do think there's value in keeping it as an error since it will make a stack trace available if it is needed for debugging. If we use a string or symbol etc, and we accidentally miss converting it somehow there is no way to trace it back to where it was created. 

Also, we _could_ set this expiry error as the cause of the error that we convert to, not the most useful but it would identify where the timeout was created (ex. in selectServer or at a higher layer for CSOT) again just for debugging not really a user-facing feature.

Reply:
I agree that for future debugging purposes, it would be useful to keep this as an error.